src/eliater/frontdoor_backdoor_discrete/sars_cov2_discrete.py

```diff
@@ -0,0 +1,240 @@
+
+    beta0_egfr = -1.9
+    beta_egf_to_egfr = 0.03
+    beta_u1_to_egfr = 0.05
+    beta_u2_to_egfr = 0.02
+    beta_u3_to_egfr = 0.04
+    beta_gefi_to_egfr = -0.08  # negative
+    if Variable("EGFR") in treatments:
+        egfr = np.full(num_samples, treatments[Variable("EGFR")])
+    else:
+        p = _r_exp(
+            -beta0_egfr
+            - egf * beta_egf_to_egfr
+            - u_il6_stat_egfr * beta_u1_to_egfr
+            - u_tnf_egfr * beta_u2_to_egfr
+            - u_egf_egfr * beta_u3_to_egfr
+            - gefi * beta_gefi_to_egfr
+        )
+        egfr = generator.binomial(1, p, size=num_samples)
+
+    beta0_prr = -1.4
+    beta_sars_cov2_to_prr = 0.05
+    beta_u_to_prr = 0.02
+    loc_prr = _r_exp(-beta0_prr - sars_cov2 * beta_sars_cov2_to_prr - u_prr_nfkb * beta_u_to_prr)
+    prr = generator.binomial(n=1, p=loc_prr, size=num_samples)
+
+    beta0_nfkb = -1.8
+    beta_prr_to_nfkb = 0.01
+    beta_u_to_nfkb = 0.02
+    beta_egfr_to_nfkb = 0.7
+    beta_tnf_to_nfkb = 0.01
+    loc_nfkb = _r_exp(
+        -beta0_nfkb
+        - prr * beta_prr_to_nfkb
+        - u_prr_nfkb * beta_u_to_nfkb
+        - egfr * beta_egfr_to_nfkb
+        - tnf * beta_tnf_to_nfkb
+    )
+    nfkb = generator.binomial(n=1, p=loc_nfkb, size=num_samples)
+
+    beta0_il6_stat3 = -1.6
+    beta_u_to_il6_stat3 = -0.05
+    beta_sil6r_to_il6_stat3 = 0.04
+    loc_il6_stat3 = _r_exp(
+        -beta0_il6_stat3 - u_il6_stat_egfr * beta_u_to_il6_stat3 - sil6r * beta_sil6r_to_il6_stat3
+    )
+    il6_stat3 = generator.binomial(n=1, p=loc_il6_stat3, size=num_samples)
+
+    beta0_il6_amp = -1.1
+    beta_nfkb_to_il6_amp = -5
+    beta_il6_stat3_to_il6_amp = 0.03
+    loc_il6_amp = _r_exp(
+        -beta0_il6_amp - nfkb * beta_nfkb_to_il6_amp - il6_stat3 * beta_il6_stat3_to_il6_amp
+    )
+    il6_amp = generator.binomial(n=1, p=loc_il6_amp, size=num_samples)
+
+    beta0_cytok = -1.1
+    beta_il6_amp_tocytok = 4
+    loc_cytok = _r_exp(-beta0_cytok - il6_amp * beta_il6_amp_tocytok)
+    cytok = generator.binomial(n=1, p=loc_cytok, size=num_samples)
+
+    data = {
+        "SARS_COV2": sars_cov2,
+        "ACE2": ace2,
+        "Ang": ang,
+        "AGTR1": agtr1,
+        "ADAM17": adam17,
+        "Toci": toci,
+        "Sil6r": sil6r,
+        "EGF": egf,
+        "TNF": tnf,
+        "Gefi": gefi,
+        "EGFR": egfr,
+        "PRR": prr,
+        "NFKB": nfkb,
+        "IL6STAT3": il6_stat3,
+        "IL6AMP": il6_amp,
+        "cytok": cytok,
+    }
+    df = pd.DataFrame(data)
+    return df
+
+
+sars_large_example = Example(
+    name="SARS-CoV-2 large Graph",
+    reference="Mohammad-Taheri, S., Zucker, J., Hoyt, C. T., Sachs, K., Tewari, V., Ness, R., & Vitek, O. 2022."
+    "Do-calculus enables estimation of causal effects in partially observed biomolecular pathways."
+    "- Bioinformatics, 38 (Supplement_1),i350-i358.",
+    description="In this example EGFR is generated as a binary value. Hence, if you want to intervene on it, please"
```

Comment:
more detail please
1. Why does this example exist
2. Does it extend previous examples?
3. Is binary realistic? Why is this useful? Compare to a continuous version for this graph

Reply:
I generated this binary version because currently we only support binary treatment when we want to use non-parametric approaches to estimate ATE. We can generate the treatment as binary and rest of the variables as continuous, but since we don't have a measure of verifying the mutual consistency of the network with the mixed-type data (one column binary, and the rest continuous, I decided to generate all the data as binary.